libc-bottom-half/sources/__environ.c

```diff
@@ -1,59 +1,63 @@
 #include <unistd.h>
 #include <stdlib.h>
+#include <sysexits.h>
 #include <wasi/api.h>
 #include <wasi/libc.h>
-#include <wasi/libc-internal.h>
 
 static char *empty_environ[1] = { NULL };
 char **__environ = empty_environ;
 extern __typeof(__environ) _environ __attribute__((weak, alias("__environ")));
 extern __typeof(__environ) environ __attribute__((weak, alias("__environ")));
 
-// This function is referenced by a weak symbol in crt1.c, and we define
-// it here in the same source file as __environ, so that this function is
-// linked in iff environment variable support is used.
-__wasi_errno_t __wasilibc_populate_environ(void) {
+// We define this function here in the same source file as __environ, so that
+// this function is called in iff environment variable support is used.
+__attribute__((constructor(0)))
```

Comment:
I'm not sure about the 0 here.  Presumably we want to reserve some number range for even lower level stuff?  (such as stuff generated by the compiler.. although I guess the linker can put any self-generated stuff before 0 if it chooses).

Reply:
Makes sense. Offhand it's hard to predict what we'd want. The reserved range is 0-100, so I'll set it to 50 to leave room on both sides.